### PR TITLE
Fix response builder

### PR DESCRIPTION
**Pull request by knowlsie**
A pull request for Issue https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues/89. This fixes `this.response` so its response value is usable, and adds documentation for how to use it.

### CODE REVIEW

File: lib/alexa.js [hunk 1]
```diff
@@ -367,6 +367,9 @@ function ResponseBuilder(self) {
 
                 responseObject.response.directives = [audioPlayerDirective];
                 return this;
+            },
+            'setResponse': function () {
```
Comment (KrishFuriaAmazon) [inline lib/alexa.js]: You don't need to setResponse when responseObject is already referencing self.response. You can just do:
```javascript
this.response.speak('foo').listen('bar');
this.emit(':responseReady');
```
And you will get the response!

Comment (KrishFuriaAmazon) [inline lib/alexa.js]: And you can access response value using `this.handler.response`.

**Comment by knowlsie** [comment]
Odd, I did some testing, and it definitely wasn’t changing this.handler.response. I’m working on some other things right now, but I’ll have a look into this...

File: lib/alexa.js (hunk 1 above)
Comment (knowlsie) [inline lib/alexa.js]: Odd, I did some testing, and it definitely wasn’t changing this.handler.response. I’m working on some other things right now, but I’ll have a look into this...

Comment (knowlsie) [inline lib/alexa.js]: My bad, it was specific to the way I was testing, the moment I upload to Lambda it worked... If you still want the docs, I'm will update those.

Comment (KrishFuriaAmazon) [inline lib/alexa.js]: Sure, since you have already put an effort in updating doc, why not get them in. :)

Comment (knowlsie) [inline lib/alexa.js]: Would now be a good time to add some dialog directive functionality to `ResponseBuilder`?  Or are you guys no longer adding features to `this.response` in favour of the more standard ways of sending response (i.e. `:ask` and `:tell`)?

Comment (KrishFuriaAmazon) [inline lib/alexa.js]: Yeah, I think we are in favor of using `:ask`, `:tell` or `:delegate`. But don't know how many devs would want to use responseBuilder over these! If devs prefer ResponseBuilder then yes we can consider adding dialog directives!

Comment (knowlsie) [inline lib/alexa.js]: Yeah, I personally prefer the usual `:ask`, `:tell` etc. interface too.

**Pull request merged**